### PR TITLE
Add authorization rules

### PR DESCRIPTION
**Pull request by zairdon20**
In this milestone I made sure that I did the following:

Installed CanCanCan in my project.
- Added a `role` column to the users table. 
- A user can delete a post if it is theirs or if they have an admin role (column `role` has value `"admin"`). Used CanCanCan for this authorization.
    - I implemented the post deleting functionality. Added the "Delete" button to the view and made sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.
    - implemented the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      can :read, :all
+      can :create, Post
+      can :create, Comment
+      can :create, Like
+
+      can :destroy, Post do |post|
+        post.author_id == user.id
+      end
+      can :destroy, Comment do |comment|
+        comment.author_id == user.id
```
Comment (Gambit142) [inline app/models/ability.rb:18]: - [ ] (Optional) You can use this shorthand syntax:

```
if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
         end

```

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -31,6 +31,16 @@ def show
     @post = User.find(params[:user_id]).posts.find(params[:id])
   end
```
Comment (Gambit142) [inline app/controllers/posts_controller.rb]: - [ ] Kindly use this line of code `  load_and_authorize_resource` before your `index` action in the `Posts` controller. This ensures your application uses `CanCanCan` for Post authorization. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @zairdon20 @farouqdaif1 👋, 

Good job so far!

## Highlights
- Installed `cancancan` gem 💯 
- Has `role` column in `Users` table 💯 
- Has _**CanCanCan**_ `ability` class 💯 
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- It would be nice to have a button that leads to the page for creating a comment on the `post#index` page. This button should be accessible to any user to comment on a post whether it is their post or not. 
- You can also consider making any user see existing comments for a particular post. 👍 
- Kindly check the optional comment under the review also. 👍  

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (zairdon20) [reply app/controllers/posts_controller.rb]: Thanks I will certainly Implement it

**Review by uwadonat — APPROVED** [review]
Hi **@Team,** 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
cong


# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**